src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -156,6 +156,16 @@ pub extern "C" fn val_for(key: Key) -> Handle {
   externs::val_for(&key).into()
 }
 
+// Like PyResult, but for values that aren't Python values.
+// throw_handle will be set iff is_throw, otherwise accessing it will likely segfault.
+// raw_pointer will be set iff !is_throw, otherwise accessing it will likely segfault.
+#[repr(C)]
+pub struct RawResult {
```

Comment:
Shoot, I know you pinged me about this, but I realized looking at this that cbindgen will actually correctly generate code for an `enum`: we use it for https://github.com/pantsbuild/pants/blob/714fdb70aff66eb835c32a1fa33e5485616671df/src/rust/engine/src/externs.rs#L480-L492, with usage here: https://github.com/pantsbuild/pants/blob/714fdb70aff66eb835c32a1fa33e5485616671df/src/python/pants/engine/native.py#L462-L468

Also fine with a TODO to change both this and PyResult at the same time.

Reply:
Heh, thanks for trying!